dotnet-interactive-vscode/src/interactiveNotebook.ts

```diff
@@ -0,0 +1,171 @@
+import { InteractiveClient } from "./interactiveClient";
+import { CommandFailed, StandardOutputValueProduced, ReturnValueProduced } from "./events";
+import { RawNotebookCell } from "./interfaces";
+import { CellDisplayOutput, CellErrorOutput, CellOutput, CellOutputKind, CellStreamOutput } from "./interfaces/vscode";
+
+export const editorLanguages = ['csharp', 'fsharp', 'html', 'javascript', 'markdown', 'powershell'];
```

Comment:
>editorLanguages [](start = 13, length = 15)

This file looks like it reimplements a bunch of the parsing in .NET Interactive. It's simplistic in that several of the language identifiers have aliases, and this list can be extended within the kernel.

Reply:
The line you highlighted is necessary, because that's the allowed list of languages in a VS Code cell that we can switch between.

As for the parsing below, we can certainly leverage the submission splitter (because that's what this really) is.  Would you prefer we replace the parsing in this file with a command:

``` js
SubmitCode('SplitSubmissionForNotebook-names-are-hard', fileContents)

=>

{
  eventType: 'NotebookSplitResult',
  event: {
    cells: [
        {
          language: 'csharp',
          contents: [
            'var x = 1;',
            'Console.WriteLine(x)'
          ]
        },
        ...
    ]
  }
}
```